distribution.md

```diff
@@ -0,0 +1,180 @@
+
+A buildpack reference in `order` MUST either be:
+
+1. A reference to a buildpack inside of the blob using a `path` field ,
+2. A reference to a buildpack outside of the blob using an `id` and `version` field. 
+
+### Buildpackage
+
+A buildpackage MUST exist as either an OCI image on an image registry, an OCI image in a Docker daemon, or a `.cnb` file.
+
+A `.cnb` file MUST be an uncompressed tar archive containing an OCI image. Its file name SHOULD end in `.cnb`.
+
+Each FS layer blob in the buildpackage MUST contain a single buildpack blob and at least one symlink.
+
+A symlink MUST be created for each buildpack version that the blob is assembled to support.
+
+```
+/cnb/blobs/<sha256 checksum of buildpack blob tgz>/
+/cnb/by-id/<buildpack ID1>/<buildpack version> -> /cnb/blobs/<sha256 checksum of blob tgz>/
+/cnb/by-id/<buildpack ID2>/<buildpack version> -> /cnb/blobs/<sha256 checksum of blob tgz>/<subdir>/
+...
+```
+
+A buildpack ID, buildpack version, and at least one stack MUST be provided in the OCI image config as a Label.
+
+Label: `io.buildpacks.buildpackage.metadata`
+```json
+{
+  "id": "<entrypoint buildpack ID>",
+  "version": "<entrypoint buildpack version>",
+  "stacks": [
+    {
+      "id": "<stack ID>",
+      "mixins": ["<mixin name>"]
+    }
+  ]
+}
+```
+
+The buildpack ID and version MUST match a buildpack provided by a layer blob.
+
+For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed stacks.
+
+For each listed stack, all associated buildpacks MUST be a candidate for detection when the entrypoint buildpack ID and version are selected.
+
+Each stack ID MUST only be present once.
+For a given stack, the `mixins` list MUST enumerate mixins such that no included buildpacks are missing a mixin for the stack.
+
+Fewer stack entries as well as additional mixins for a stack entry MAY be specified.
+
+## Data Format
+
+### buildpack.toml (TOML)
+
+```toml
+[buildpack]
+id = "<buildpack ID>"
+name = "<buildpack name>"
+version = "<buildpack version>"
+
+[[order]]
+[[order.group]]
+id = "<buildpack ID>"
+version = "<buildpack version>"
+optional = false
+path = "<path to buildpack>"
+
+[[stacks]]
+id = "<stack ID>"
+mixins = ["<mixin name>"]
+build-images = ["<build image tag>"]
+run-images = ["<run image tag>"]
+
+[metadata]
+# buildpack-specific data
+```
+
+If an order is specified, then `stacks` MUST not be specified.
+
+A buildpack reference inside of a `group` MUST either contain an `id` and `version` or a relative `path`.
+
+Buildpack authors MUST choose a globally unique ID, for example: "io.buildpacks.ruby".
+
+The buildpack ID:
+- MUST only contain numbers, letters, and the characters `.`, `/`, and `-`.
+- MUST NOT be `config` or `app`.
+- MUST NOT be identical to any other buildpack ID when using a case-insensitive comparison.
+
+Stack authors MUST choose a globally unique ID, for example: "io.buildpacks.mystack".
```

Comment:
"globally unique" might be a tad confusing given that we encourage users to re-use a given stack ID to denote ABI compatibility. Wonder if there's a way to expand on that a bit here.

Reply:
We ran into a similar issue when documenting the basics of a custom stack. You can see what we did here: https://buildpacks.io/docs/using-pack/stacks/#creating-custom-stacks

Reply:
A stack is a contract, not a specific set of images. The author of that contract must choose a globally unique ID that those creating stack images following that contract can use.

I think we should change the wording from "custom stack" to "stack images" in the docs. (The platform spec should probably also be clarified.)